cms/djangoapps/contentstore/management/commands/force_publish.py

```diff
@@ -0,0 +1,43 @@
+"""
+Script for force publishing a course
+"""
+from django.core.management.base import BaseCommand, CommandError
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from .prompt import query_yes_no
+
+# To run from command line: ./manage.py cms force_publish course-v1:org+course+run
+
+
+class Command(BaseCommand):
+    """Force publish a course"""
+    help = '''
+    Force publish a course. Takes one arguments:
+    <course_id>: the course id of the course you want to publish forcefully
+    '''
+
+    def handle(self, *args, **options):
+        "Execute the command"
+        if len(args) != 1:
+            raise CommandError("force_publish requires 1 argument: <course_id>")
+
+        try:
+            course_key = CourseKey.from_string(args[0])
+        except InvalidKeyError:
+            raise CommandError("Invalid course key.")
+
+        if not modulestore().get_course(course_key):
+            raise CommandError("Course not found.")
+
+        # for now only support on split mongo
+        owning_store = modulestore()._get_modulestore_for_courselike(course_key)  # pylint: disable=protected-access
+        if hasattr(owning_store, 'force_publish_course'):
+            if query_yes_no("Are you sure to publish the {0} course forcefully?".format(course_key), default="no"):
+                if owning_store.force_publish_course(course_key, ModuleStoreEnum.UserID.mgmt_command):
+                    print "Published the course '{0}' forcefully.".format(course_key)
+                else:
+                    print "Course '{0}' already in published state.".format(course_key)
+        else:
```

Comment:
I look forward to using this command (instead of manually editing documents in Mongo).  A few things that would make life easier for devops.

If it printed the draft-branch and published-branch data before updating (so if something goes wrong, I have the old id at hand to set it back).

Additionally, if it took two args (we often do <courseid> commit in these commands) so I can run it once as a noop that prints "Would update published branch <foo> to draft branch <bar>" and similar with commit prints "Updated published branch <foo> to draft branch <bar>".

I also see that you handle the published == draft case by doing nothing, it would be great to report that back so we know why it did nothing.  Otherwise, I look forward to using this code.